pkg/agent/agent.go

```diff
@@ -67,18 +69,44 @@ type proc interface {
 	KillProcess() error
 }
 
-func New(log logging.Logger, kube kubernetes.Interface, plat platform.Platform, nodeName string) (*Agent, error) {
+func New(log logging.Logger, kube kubernetes.Interface, nodeName string) (*Agent, error) {
 	if nodeName == "" {
 		return nil, errors.New("nodeName must be provided for Agent to manage")
 	}
-	var nodeclient corev1.NodeInterface
-	if kube != nil {
-		nodeclient = kube.CoreV1().Nodes()
+	var (
+		nodeclient corev1.NodeInterface
+		platform   platform.Platform
+	)
+	nodeclient = kube.CoreV1().Nodes()
```

Comment:
Now that `nodeclient` is always set up, this can be reduced to (with `platform` moved below):

```suggestion
	nodeclient = kube.CoreV1().Nodes()
```